tempesta_fw/t/unit/test_sched_hash.c

```diff
@@ -43,96 +43,90 @@
 #include "sched_helper.h"
 #include "test.h"
 
-char req_str1[] = "GET / HTTP/1.1\r\nhost:host1\r\n\r\n";
-char req_str2[] = "GET / HTTP/1.1\r\nhost:host2\r\n\r\n";
-char req_str3[] = "GET / HTTP/1.1\r\nhost:host3\r\n\r\n";
-char req_str4[] = "GET / HTTP/1.1\r\nhost:host4\r\n\r\n";
-char *req_strs[] = {
-	req_str1,
-	req_str2,
-	req_str3,
-	req_str4,
+static char *req_strs[] = {
+	"GET / HTTP/1.1\r\nhost:host1\r\n\r\n",
+	"GET / HTTP/1.1\r\nhost:host2\r\n\r\n",
+	"GET / HTTP/1.1\r\nhost:host3\r\n\r\n",
+	"GET / HTTP/1.1\r\nhost:host4\r\n\r\n",
 };
-size_t req_strs_size = sizeof(req_strs) / sizeof(req_strs[0]);
 
-TEST(tfw_sched_hash, sg_empty)
+static TfwMsg *sched_hash_get_scheduler_arg(size_t connection_type);
+
+static void
+sched_hash_free_scheduler_arg(TfwMsg *msg)
 {
-	int i, j;
+	test_req_free((TfwHttpReq *)msg);
+}
 
-	TfwSrvGroup *sg = test_create_sg("test", "hash");
+static struct TestSchedHelper sched_helper_hash = {
+	.sched = "hash",
+	.conn_types = ARRAY_SIZE(req_strs),
+	.get_sched_arg = &sched_hash_get_scheduler_arg,
+	.free_sched_arg = &sched_hash_free_scheduler_arg,
+};
 
-	for (i = 0; i < req_strs_size; ++i) {
-		for (j = 0; j < 3; ++j) {
-			TfwConnection *conn;
-			TfwHttpReq *req = test_req_alloc(strlen(req_strs[i]));
+static TfwMsg *
+sched_hash_get_scheduler_arg(size_t connection_type)
+{
+	TfwHttpReq *req = NULL;
 
-			tfw_http_parse_req(req, req_strs[i],
-					   strlen(req_strs[i]));
+	BUG_ON(connection_type >= sched_helper_hash.conn_types);
 
-			conn = sg->sched->sched_srv((TfwMsg *)req, sg);
-			EXPECT_TRUE(conn == NULL);
+	req = test_req_alloc(strlen(req_strs[connection_type]));
+	tfw_http_parse_req(req, (unsigned char *) req_strs[connection_type],
+					   strlen(req_strs[connection_type]));
```

Comment:
Not a problem in this particular case, but more often than not long names are a burden as code statements don't fit a single line. `conn_type` is simpler and shorter without sacrificing the meaning.